### ISSUE TITLE
Zero-Shot Performance of the Huggingface Checkpoint

### ISSUE THREAD

**Issue by Xingrun-Xing2** [facebookresearch/MobileLLM#17]
Hi, does anyone know how to evaluate zero-shot performance in the paper? My test script is as follows, but I get a random result. I wonder how to load this checkpoint correctly from huggingface.

![image](https://github.com/user-attachments/assets/06caaeae-0cd5-497f-9a06-44d7a6d929ae)

```python
from transformers import AutoModelForCausalLM, AutoTokenizer
import lm_eval
from lm_eval import utils as lm_eval_utils
from lm_eval.api.registry import ALL_TASKS
from lm_eval.models.huggingface import HFLM

tokenizer = AutoTokenizer.from_pretrained("facebook/MobileLLM-125M", use_fast=False)
model = AutoModelForCausalLM.from_pretrained("facebook/MobileLLM-125M", trust_remote_code=True)

model.to("cuda")

tokenizer.add_special_tokens(
    {
        "eos_token": "</s>",
        "bos_token": "<s>",
        "unk_token": "<unk>",
    }
)


task_manager = lm_eval.tasks.TaskManager()
hflm = HFLM(pretrained=model, tokenizer=tokenizer, batch_size=16)

results = lm_eval.simple_evaluate(hflm, tasks=["piqa","hellaswag","winogrande","arc_challenge","arc_easy","boolq","openbookqa"], batch_size=16, num_fewshot=0, task_manager=task_manager)['results']


metric_vals = {task: round(result.get('acc_norm,none', result['acc,none']), 4) for task, result in results.items()}
metric_vals['acc_avg'] = round(sum(metric_vals.values()) / len(metric_vals.values()), 4)

print("Evaluation results:")
for task, score in metric_vals.items():
    print(f"{task}: {score}")

print(f"Overall accuracy (average): {metric_vals['acc_avg']}")

**Comment by liuzechun** [comment]
Hello, thank you for your question. Huggingface safetersor model conversion is incorrect...the original pytorch_model.bin is correct. Now that we have removed the Huggingface safetersor it should work. If you have any other questions, please let me know!

**Comment by Xingrun-Xing2** [comment]
Really appreciate this work. It helps a lot!
